### PR TITLE
test csi driver in vm mode

### PR DESCRIPTION
**Pull request by antonmyagkov**
1. Test NodeStageVolume
2. Test NodeUnstageVolume
3. Test VM Mode
4. Set VhostEnabled by default
5. Test publishing volume twice on the same node in VM Mode

### CODE REVIEW

File: cloud/blockstore/tools/csi_driver/client/main.go [hunk 1]
```diff
@@ -189,6 +189,70 @@ func newDeleteVolumeCommand(endpoint *string) *cobra.Command {
...
+		Use:   "stagevolume",
+		Short: "Send stage volume request to the CSI node",
+		Run: func(cmd *cobra.Command, args []string) {
+			ctx, cancelFunc := context.WithTimeout(
+				context.Background(),
+				120*time.Second,
+			)
+			defer cancelFunc()
+			client, err := newNodeClient(ctx, *endpoint)
+
```
Comment (jkuradobery) [inline cloud/blockstore/tools/csi_driver/client/main.go:204]: Extra line break, err  != nil is usually padded towards the call which created the error.

File: cloud/blockstore/tests/csi_driver/test.py [hunk 2]
```diff
@@ -360,6 +392,7 @@ def test_csi_sanity_nbs_backend(mount_path, volume_access_type):
 
         params_file = Path(os.getcwd()) / "params.yaml"
         params_file.write_text(f"backend: {backend}")
```
Comment (jkuradobery) [inline cloud/blockstore/tests/csi_driver/test.py:394]: Let us use yaml.safe_dump here, shall we?

File: cloud/blockstore/tools/csi_driver/client/main.go [hunk 3]
```diff
@@ -189,6 +189,70 @@ func newDeleteVolumeCommand(endpoint *string) *cobra.Command {
...
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			volumeContext := map[string]string{
+				"instanceId": "example-instance-id",
+			}
+			writerCap := csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER
```
Comment (jkuradobery) [inline cloud/blockstore/tools/csi_driver/client/main.go:212]: `accessMode`?

File: cloud/blockstore/tools/csi_driver/client/main.go (hunk 1 above)
Comment (antonmyagkov) [reply cloud/blockstore/tools/csi_driver/client/main.go]: Fixed here and in other places

File: cloud/blockstore/tools/csi_driver/client/main.go (hunk 3 above)
Comment (antonmyagkov) [reply cloud/blockstore/tools/csi_driver/client/main.go]: Renamed here and in other places

File: cloud/blockstore/tests/csi_driver/test.py (hunk 2 above)
Comment (antonmyagkov) [reply cloud/blockstore/tests/csi_driver/test.py]: Done

File: cloud/blockstore/tests/csi_driver/test.py [hunk 4]
```diff
@@ -346,20 +378,26 @@ def test_nbs_csi_driver_volume_stat():
...
         mount_dir = Path(mount_path)
         mount_dir.parent.mkdir(parents=True, exist_ok=True)
 
+        volume_context = {
+            "backend": backend,
+            "instanceId": "test-instance-id",
+        }
         params_file = Path(os.getcwd()) / "params.yaml"
-        params_file.write_text(f"backend: {backend}")
+        with open(params_file, 'w') as f:
```
Comment (jkuradobery) [inline cloud/blockstore/tests/csi_driver/test.py:399]: with params_file.open('w') or even better params_file.write_text(yaml.safe_dump(volume_context)). And you can inline volume_context then. All that can be made into a one logical line. Also, Path("params.yaml") would achieve the same result as os.getcwd() / "params.yaml", so the code can be simplified even more.

File: cloud/blockstore/tests/csi_driver/test.py [hunk 5]
```diff
@@ -419,6 +458,43 @@ def test_node_volume_expand():
...
+        pod_id2 = "deadbeef2"
+        env.csi.create_volume(name=volume_name, size=volume_size)
+        env.csi.stage_volume(volume_name)
+        env.csi.publish_volume(pod_id1, volume_name, pod_name1)
+        env.csi.publish_volume(pod_id2, volume_name, pod_name2)
+    except subprocess.CalledProcessError as e:
+        log_called_process_error(e)
+        raise
+    finally:
+        try:
```
Comment (jkuradobery) [inline cloud/blockstore/tests/csi_driver/test.py:489]: Let us make a simple context manager for that:
```
@contextlib.contextmanager
def called_process_error_logged():
    try:
         yield
    except subprocess.CalledProcessError as e:
           log_called_process_error(e)
```

Comment (antonmyagkov) [reply cloud/blockstore/tests/csi_driver/test.py]: Done

**Review by jkuradobery — APPROVED** [review]

**Review by EvgeniyKozev — APPROVED** [review]

**Review by jkuradobery — APPROVED** [review]

**Review by EvgeniyKozev — APPROVED** [review]

**Pull request merged**